src/halmos/cheatcodes.py

```diff
@@ -407,20 +413,20 @@ def create_int(ex, arg, **kwargs):
         raise HalmosException(f"bitsize larger than 256: {bits}")
 
     name = name_of(extract_string_argument(arg, 1))
-    return ByteVec(int256(create_generic(ex, bits, name, f"int{bits}")))
+    return ByteVec(int256(create_any(ex, bits, name, f"int{bits}")))
 
 
 def create_int256(ex, arg, **kwargs):
     name = name_of(extract_string_argument(arg, 0))
-    return ByteVec(create_generic(ex, 256, name, "int256"))
+    return ByteVec(create_any(ex, 256, name, "int256"))
 
 
 def create_bytes(ex, arg, **kwargs):
-    byte_size = int_of(
-        extract_bytes(arg, 4, 32), "symbolic byte size for halmos.createBytes()"
-    )
+    err = "symbolic byte size for halmos.createBytes()"
+    byte_size = int_of(extract_bytes(arg, 4, 32), err)
+    bits = byte_size * 8
     name = name_of(extract_string_argument(arg, 1))
-    symbolic_bytes = create_generic(ex, byte_size * 8, name, "bytes")
+    symbolic_bytes = create_any(ex, bits, name, "bytes") if bits else ByteVec()
```

Comment:
an alternative approach is to add the empty byte logic inside create_any() rather than here.  though, it's unclear to me which is necessarily better.

if we decide to keep it here, then we also need to add it to create_string() below, which is basically the same as create_bytes() except for the error message.  indeed, it would be better to factor out the common logic to avoid code duplication.

Reply:
https://github.com/a16z/halmos/pull/488